services/open-vsx/open-vsx-base.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const Joi = require('joi')
+const { optionalNonNegativeInteger } = require('../validators')
+const { BaseJsonService, NotFound } = require('..')
+
+const extensionQuerySchema = Joi.object({
+  error: Joi.string(),
+  version: Joi.string().when('error', {
+    is: Joi.exist(),
+    then: Joi.forbidden(),
+    otherwise: Joi.required(),
+  }),
+  timestamp: Joi.string().isoDate().when('error', {
+    is: Joi.exist(),
+    then: Joi.forbidden(),
+    otherwise: Joi.required(),
+  }),
+  downloadCount: optionalNonNegativeInteger,
+  reviewCount: optionalNonNegativeInteger,
+  averageRating: Joi.number().when('reviewCount', {
+    is: Joi.exist(),
+    then: Joi.number().max(5),
+    otherwise: Joi.forbidden(),
+  }),
+}).required()
+
+module.exports = class OpenVSXBase extends BaseJsonService {
+  static keywords = [
+    'ovsx',
+    'open-vsx',
+    'ovsx-marketplace',
+    'open-vsx-marketplace',
+  ]
+
+  static defaultBadgeData = {
+    label: 'open vsx',
+    color: 'blue',
+  }
+
+  async fetch({ namespace, extension, version }) {
+    const baseUrl = 'https://open-vsx.org/api'
+    const options = {
+      method: 'GET',
+    }
+
+    const url = version
+      ? `${baseUrl}/${namespace}/${extension}/${version}`
+      : `${baseUrl}/${namespace}/${extension}`
+
+    return this._requestJson({
+      schema: extensionQuerySchema,
+      url,
+      options,
+      errorMessages: {
+        400: 'invalid extension id',
+      },
+    })
```

Comment:
missed this on the first pass, but think we can express this all a bit more succinctly, e.g.
```suggestion
    return this._requestJson({
      schema: extensionQuerySchema,
      url: `https://open-vsx.org/api/${namespace}/${extension}/${version || ''}`,
      errorMessages: {
        400: 'invalid extension id',
      },
    })
```